tools/debug.js

```diff
@@ -35,7 +35,12 @@
   });
 
   function toStr(obj) {
-    let str = JSON.stringify(obj || {});
+    let str;
+    try {
+      str = JSON.stringify(obj || {});
+    } catch (e) {
+      str = "<...>";
```

Comment:
@ged-odoo 
Might be ambiguous with `...` in case the string is too long
I'd go for for something explicit e.g. `<JSON error>`

Reply:
I agree